src/picom.c

```diff
@@ -680,6 +680,47 @@ static struct managed_win *paint_preprocess(session_t *ps, bool *fade_running) {
 		if (was_painted && w->mode != mode_old) {
 			w->reg_ignore_valid = false;
 		}
+
+		// Transition
+		if (w->transition_direction && w->transition_time >= 0.0f && w->transition_time <= 1.0f) {
+			double transition = ps->o.transition_timing_function(w->transition_time);
+			
+			w->transition_time += ps->o.transition_step; 
+            if (w->transition_time > 1.0f)
+                transition = 1.0f;
+
+			add_damage_from_win(ps, w);
+			unsigned int direction = w->transition_direction;
+
+			// Smart direction
+			if (direction == TRANSITION_DIR_SMART_X || direction == TRANSITION_DIR_SMART_Y) {
+				bool wide_enough = w->g.width > 80 * ps->root_width / 100;
```

Comment:
Does it make sense to make this 80% configurable, or would it be too verbose?

Reply:
I think that would be much verbose.

Reply:
Fair enough